config/routes.rb

```diff
@@ -292,21 +292,19 @@
       get "/signup" => "people#new", :as => :sign_up
       get '/people/auth/:provider/setup' => 'sessions#facebook_setup' #needed for devise setup phase hook to work
 
-      resources :people, :path => "", :only => :show, :constraints => { :id => /[_a-z0-9]+/ }
+      resources :people, param: :username, :path => "", :only => :show, :constraints => { :username => /[_a-z0-9]{3,20}/ }
```

Comment:
Good question. And the answer is no.
From the Rails [documentation](http://guides.rubyonrails.org/routing.html#segment-constraints):

> :constraints takes regular expressions with the restriction that regexp anchors can't be used.
and
> However, note that you don't need to use anchors because all routes are anchored at the start.